src/cargo/ops/cargo_rustc/fingerprint.rs

```diff
@@ -173,25 +175,36 @@ fn deserialize_deps<'de, D>(d: D) -> Result<Vec<(String, Arc<Fingerprint>)>, D::
 enum LocalFingerprint {
     Precalculated(String),
     MtimeBased(MtimeSlot, PathBuf),
+    EnvBased(VarSlot, String),
 }
 
 struct MtimeSlot(Mutex<Option<FileTime>>);
+struct VarSlot(Mutex<Option<String>>);
```

Comment:
Huh now I'm not even sure how I ended up having it implemented this way... In any case I believe you're 100% correct!